include/ESA.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef ESA_HPP
+#define ESA_HPP
+
+#include <limits>
+#include <map>
+#include "stls.hpp"
+
+// -----------------------------------------------------------------
+// Solver for the ESA scheme
+// -----------------------------------------------------------------
+
+class ESA : public Stls {
+
+protected:
+
+  // Function to perform the ESA scheme
+  void doESA();
+
+public:
+
+  // ESA constructor (inherited from Stls)
+  ESA(const StlsInput& input);
+  // Compute ESA scheme
+  int compute();
+  // Funtion for the ESA SLFC
+  void computeSlfc();
```

Comment:
Make computeSlfc and all the methods to compute the derivatives private. They are never called from outside the ESA class